lca-cli/postpivot/postpivot.go

```diff
@@ -649,21 +691,40 @@ func (p *PostPivot) copyNMConnectionFiles(source, dest string) error {
 	return nil
 }
 
+// physicalNetworkConfigurations is on charge of configuring host network prior installation process.
+// 1. Copying provided nmconnection files to NM sysconnections to setup network provided by user
+// 2. Apply provided NMstate configurations
+// 3. We should restart NM after adding connection files in order for sure to set static ip
+func (p *PostPivot) physicalNetworkConfigurations(seedReconfiguration *clusterconfig_api.SeedReconfiguration) error {
+	if err := p.copyNMConnectionFiles(
+		path.Join(p.workingDir, common.NetworkDir, "system-connections"), nmConnectionFolder); err != nil {
+		return err
+	}
+
+	if err := p.applyNMStateConfiguration(seedReconfiguration); err != nil {
+		return err
+	}
+
+	if _, err := p.ops.SystemctlAction("restart", nmService); err != nil {
+		return fmt.Errorf("failed to restart network manager service, err %w", err)
+	}
+	return nil
+}
+
 // networkConfiguration is on charge of configuring network prior installation process.
 // This function should include all network configurations of post-pivot flow.
 // It's logic currently includes:
-// 1. Copying provided nmconnection files to NM sysconnections to setup network provided by user
-// 2. Apply provided NMstate configurations
+// 1. Setup host network
+// 2. Override nodeip-hint file
 // 3. In case ip was not provided by user we should run set ip logic that can be found in setNodeIPIfNotProvided
 // 4. Override seed dnsmasq params
 // 5. Restart NM and dnsmasq in order to apply provided configurations
 func (p *PostPivot) networkConfiguration(ctx context.Context, seedReconfiguration *clusterconfig_api.SeedReconfiguration) error {
-	if err := p.copyNMConnectionFiles(
-		path.Join(p.workingDir, common.NetworkDir, "system-connections"), nmConnectionFolder); err != nil {
+	if err := p.physicalNetworkConfigurations(seedReconfiguration); err != nil {
 		return err
 	}
 
-	if err := p.applyNMStateConfiguration(seedReconfiguration); err != nil {
+	if err := p.setNodeIPHint(ctx, seedReconfiguration.NodeIP, nodeIPHintFile); err != nil {
```

Comment:
Perhaps we should do this after executing `setNodeIPIfNotProvided` ?

Reply:
they actually mutually exclusive, one will run only of ip was provided and another one will only if not but sure we can change

Reply:
we can actually in this case ip will always exists and we will just update nodeip hint with right value.